### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by iAmWaris97**
In this step, I did:

1.  Created a method called current_user in ApplicationController to make current user data available to all controllers.
2.  Created forms to perform the following functions:
3.  Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
4.  Creates a comment on behalf of the current_user (use the method that you created in your ApplicationController).

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new
+    @like.author = current_user
+    @like.post = Post.find(params[:post_id])
+    @like.save!
+    redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
+  end
```
Comment (Munsa1) [inline app/controllers/likes_controller.rb:8]: - [ ] The user is supposed to be able to add a comment and a like, however what we have here is an error when the user tries to click on each single post. Kindly fix this issue

image

Result After Clicking on Each Post
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @iAmWaris97, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Good job creating routes for creating posts :heavy_check_mark: 
Great work creting the views which accepts input for creating posts :heavy_check_mark: 
Nice work creating a route for a new comment :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/likes_controller.rb:8]: Needs attention!

Comment (iAmWaris97) [reply app/controllers/likes_controller.rb:8]: Sir I updated these files
My files do not look like this right now, and everything is working,

Comment (iAmWaris97) [reply app/controllers/likes_controller.rb:8]: > Needs attention!

Could you please check my latest commit?
I entirely updated this file

Thank you

Comment (rloterh) [reply app/controllers/likes_controller.rb:8]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Waris,  

Your project is complete! 🏁 Great job! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Posts features correctly added!
- Comments works as intended!
- Like feature added!
- Good ReadMe!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Live demo link in your ReadMe is broken, kindly endeavor to fix or remove it if this app is not yet deployed.

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (iAmWaris97) [reply app/controllers/likes_controller.rb:8]: > well resolved! ✔️

So could you please review this project?

Thank you

Comment (rloterh) [reply app/controllers/likes_controller.rb:8]: Done!
Pls, however, do not introduce new changes when your project is under review!

Cheers! 😃

**Comment by rloterh** [comment]
Hi @iAmWaris97 

I'd you to kindly confirm if your project is approved at the backend.

Standing by

**Pull request merged**